File: lib/to_mdx/parsers/open_document/dc/description.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module DublinCore
      class Description < Base
        attr_reader :description_text

        def initialize(element = nil, context = {})
          super(element, context)
          @description_text = element&.text&.strip

          if @description_text && !@description_text.empty?
            truncated = @description_text.length > 80 ? "#{@description_text[0..77]}..." : @description_text
            puts("🔍 Found presentation description: #{truncated}")

            context["description"] = @description_text
          end
        end

        def to_mdx
          ""
        end

        def to_s
          description_text || ""
        end

        register_for "dc:description"
      end
    end
  end
end

File: src/parsers/open-document/dc/description.ts
import type { ParseContext } from "../../base_element.ts";
import { BaseElement } from "../../base_element.ts";

export class Description extends BaseElement {
  private descriptionText: string;

  constructor(element: Element, context: ParseContext, parent: BaseElement | null) {
    super(element, context, parent);
    this.descriptionText = (element.textContent || "").trim();

    if (this.descriptionText) {
      const truncated =
        this.descriptionText.length > 80 ? `${this.descriptionText.slice(0, 77)}...` : this.descriptionText;
      console.log(`🔍 Found presentation description: ${truncated}`);

      context.description = this.descriptionText;
    }
  }

  toMdx(): string {
    return "";
  }

  toString(): string {
    return this.descriptionText || "";
  }

  static {
    BaseElement.registerFor.call(this, "dc:description");
  }
}
